src/main/java/org/sagebionetworks/bridge/exporter/dynamo/DynamoHelper.java

```diff
@@ -196,6 +201,57 @@ public StudyInfo getStudyInfo(String studyId) {
     }
 
     /**
+     * Helper method to generate study ids for query
+     * @param request
+     * @return
+     */
+    public Map<String, DateTime> bootstrapStudyIdsToQuery(BridgeExporterRequest request, DateTime endDateTime) {
+        // first check if it is s3 override request
+        if (StringUtils.isNotBlank(request.getRecordIdS3Override())) {
+            return ImmutableMap.of();
+        }
+
+        List<String> studyIdList = new ArrayList<>();
+
+        if (request.getStudyWhitelist() == null) {
+            // get the study id list from ddb table
+            Iterable<Item> scanOutcomes = ddbScanHelper.scan(ddbStudyTable);
+            for (Item item: scanOutcomes) {
+                studyIdList.add(item.getString(IDENTIFIER));
+            }
+        } else {
+            studyIdList.addAll(request.getStudyWhitelist());
+        }
+
+        // maintain insert order for testing
+        Map<String, DateTime> studyIdsToQuery = new HashMap<>();
+
+        for (String studyId : studyIdList) {
+            Item studyIdItem = ddbExportTimeTable.getItem(STUDY_ID, studyId);
+            if (studyIdItem != null && !request.getIgnoreLastExportTime()) {
+                DateTime lastExportDateTime = new DateTime(studyIdItem.getLong(LAST_EXPORT_DATE_TIME), timeZone);
+                if (!endDateTime.isBefore(lastExportDateTime)) {
```

Comment:
What happens if endDateTime is null? We just hit this error in production. We should add a null-check and test for it here.